delegation/src/Pos/Delegation/Logic/Common.hs

```diff
@@ -79,21 +77,9 @@ runDelegationStateAction action = do
 ----------------------------------------------------------------------------
 
 -- | Retrieves last PSK in chain of delegation started by public key
--- and resolves the passed issuer to a public key. Doesn't check that
--- user himself didn't delegate. Uses database only.
+-- and resolves the passed issuer to a public key. Uses database only.
 getDlgTransPsk
     :: (MonadDBRead m, MonadUnliftIO m)
     => StakeholderId -> m (Maybe (PublicKey, ProxySKHeavy))
-getDlgTransPsk issuer = getDlgTransitive issuer >>= \case
-    Nothing -> pure Nothing
-    Just dPk -> do
-        chain <- runDBCede $ HM.elems <$> getPskChain issuer
-        let finalPsk =
-                find (\psk -> addressHash (pskDelegatePk psk) == dPk) chain
-        let iPk = pskIssuerPk <$>
-                  find (\psk -> addressHash (pskIssuerPk psk) == issuer) chain
-        let throwEmpty = throwM $ DBMalformed $
-                sformat ("getDlgTransPk: couldn't find psk with dlgPk "%build%
-                         " and issuer "%build)
-                        dPk issuer
-        maybe throwEmpty (pure . Just) $ (,) <$> iPk <*> finalPsk
+getDlgTransPsk issuer =
+    fmap (\psk -> (pskIssuerPk psk, psk)) <$> runDBCede (dlgLastPsk issuer)
```

Comment:
I suspect you changed the semantics of this function. Previouisly it was returning a `PublicKey` such that its `addressHash` was equal to `issuer`. Now it returns `pskIssuerPk` from the last PSK in the chain.
Is my understanding correct? If it is, is this change intentional?

Reply:
It was not.